test/test-functions

```diff
@@ -1421,16 +1425,16 @@ do_test() {
         [[ -d $usrlibdir ]] && libdirs+=" $usrlibdir" && break
     done
 
+    mkdir -p "$STATEDIR"
+
     import_testdir
     import_initdir
 
     while (($# > 0)); do
         case $1 in
             --run)
                 echo "TEST RUN: $TEST_DESCRIPTION"
-                test_run
-                ret=$?
```

Comment:
I figured out that`ret` is a global variable set by `test_run`. It's a bit confusing, but I can probably live with that :-)